### ISSUE TITLE
Running averages for batch norm

### ISSUE THREAD

**Issue by axiomizer** [axiomizer/haida-go#4]
The class src.nn.batch_norm.BatchNorm should probably keep running averages, the same way that torch.nn.BatchNorm2D does.

[Docs for BatchNorm2D](https://pytorch.org/docs/stable/generated/torch.nn.BatchNorm2d.html)

I think this will need a unit test.

**Comment by axiomizer** [comment]
Implemented in 082c3f6a00a5ee6cd107c092a757e408d4cdecc2

**Comment by axiomizer** [comment]
Unit test added in 000726f4da252441f75903308d509dcd935e1fea

**Comment by axiomizer** [comment]
Unit test added in 000726f4da252441f75903308d509dcd935e1fea
